metric_learn/base_metric.py

```diff
@@ -369,6 +386,168 @@ def score(self, pairs, y):
     """
     return roc_auc_score(y, self.decision_function(pairs))
 
+  def set_default_threshold(self, pairs, y):
+    """Sets the default threshold on the given dataset.
+
+    Returns a threshold that is the mean between the similar
+    metrics mean and the dissimilar metrics mean.
+
+    See more in the :ref:`User Guide <calibration>`.
+    """
+    similar_threshold = np.mean(self.score_pairs(
+        pairs[(y == 1).ravel()]))
+    dissimilar_threshold = np.mean(self.score_pairs(
+        pairs[(y == -1).ravel()]))
+    self.threshold_ = np.mean([similar_threshold, dissimilar_threshold])
+
+  def set_threshold(self, threshold):
+    """Sets the threshold of the metric learner to the given value `threshold
+
+    See more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    threshold : float
+      The threshold value we want to set. It's a distance metric with
+      respect to which the predicted distance metric for test pairs will be
+      compared to. If they are superior to the threshold they will be
+      classified as similar (+1), and dissimilar (-1) if not.
+
+    Returns
+    -------
+    self : `_PairsClassifier`
+      The pairs classifier with the new threshold set.
+    """
+    self.threshold_ = threshold
+    return self
+
+  def calibrate_threshold(self, pairs_valid, y_valid, strategy='accuracy',
+                          threshold=None, beta=None):
```

Comment:
we could use 1 as default value for beta, as f1 score is very standard

Reply:
Agreed, that's also what is done in scikit-learn  for `precision_recall_fscore_support`
done